### PR TITLE
Update vault.py

### PR DESCRIPTION
**Pull request by jeffwarnica**
Add warning for stdin strings ending with a newline.

##### ISSUE TYPE

- Bugfix Pull Request
- Docs Pull Request
- Feature Pull Request
- Test Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by jeffwarnica** [comment]
https://github.com/ansible/ansible/issues/72229

File: lib/ansible/cli/vault.py [hunk 1]
```diff
@@ -318,6 +318,9 @@ def execute_encrypt_string(self):
             if stdin_text == '':
                 raise AnsibleOptionsError('stdin was empty, not encrypting')
 
+            if not sys.stdout.isatty() and stdin_text.endswith("\n"):
+                display.display("Warning: Provided string ends with a newline. It is unlikely this is correct. Did you mean 'echo -n'?", stderr=True)
```
Comment (bcoca) [inline lib/ansible/cli/vault.py]: use `display.warning` for warnings

**Review by bcoca — CHANGES_REQUESTED** [review]
Also add a test to avoid regressions.

File: lib/ansible/cli/vault.py (hunk 1 above)
Comment (jeffwarnica) [reply lib/ansible/cli/vault.py]: Updated to warnings, but I wasn't able to figure out how to detect a warning from the current vault tests.

File: lib/ansible/cli/vault.py [hunk 2]
```diff
@@ -318,6 +318,9 @@ def execute_encrypt_string(self):
             if stdin_text == '':
                 raise AnsibleOptionsError('stdin was empty, not encrypting')
 
+            if not sys.stdout.isatty() and stdin_text.endswith("\n"):
+                display.warning("Warning: Provided string ends with a newline. It is unlikely this is correct. Did you mean 'echo -n'?", stderr=True)
```
Comment (Akasurde) [inline lib/ansible/cli/vault.py:322]:
```suggestion
                display.warning("Provided string ends with a newline. It is unlikely this is correct. Did you mean 'echo -n'?")
```

**Comment by nitzmahone** [comment]
Closing for inactivity

**Pull request closed without merging**